### PR TITLE
Adding doctest example for `image-classification` pipeline.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/pipelines/image_classification.py [hunk 1]
```diff
@@ -34,6 +34,22 @@ class ImageClassificationPipeline(Pipeline):
...
+    ```python
+    >>> from transformers import pipeline
+
+    >>> classifier = pipeline(model="microsoft/beit-base-patch16-224-pt22k-ft22k")
+    >>> potential_words = classifier("https://huggingface.co/datasets/Narsil/image_dummy/raw/main/parrots.png")
+
+    >>> from transformers.testing_utils import nested_simplify
+
+    >>> nested_simplify(potential_words)  # The scores might vary slightly across pytorch/tensorflow versions.
+    [{'score': 0.442, 'label': 'macaw'}, {'score': 0.088, 'label': 'popinjay'}, {'score': 0.075, 'label': 'parrot'}, {'score': 0.073, 'label': 'parodist, lampooner'}, {'score': 0.046, 'label': 'poll, poll_parrot'}]
```
Comment (sgugger) [inline src/transformers/pipelines/image_classification.py:48]: You don't need to use this, doctest will only compare the floats to the precision you write them. So the test will pass without using `nested_simplify`.

**Review by sgugger — COMMENTED** [review]
Let's avoid using Transformers internal testing tools in examples. Rounding shouldn't be necessary, and even if it was I'd prefer to see it done in a dict comprehension, not by using `nested_simplify`.

File: src/transformers/pipelines/image_classification.py (hunk 1 above)
Comment (Narsil) [reply src/transformers/pipelines/image_classification.py:48]: Really ?? That's awesome, I can remove it in a lot of places then !

**Review by stevhliu — APPROVED** [review]
Looks good, thanks!
